### PR TITLE
Add `adjust!`

### PR DESCRIPTION
**Pull request by mcabbott**
Now that `Leaf` is mutable, perhaps it is useful to be able to change `eta` only for some parts of the model.

PR also solves the following bug: `adjust` doesn't copy the momenta, and thus they remain shared with the previous object. 

### PR Checklist

- [x] Tests are added
- [x] Documentation, if applicable

### CODE REVIEW

**Comment by mcabbott** [comment]
We should do this, IMO. It fits with `freeze!`, and updating e.g. https://fluxml.ai/Flux.jl/dev/tutorials/2021-02-07-convnet/ to explicit parameters would like to use it.

Maybe `adjust` should in fact be marked deprecated, in favour of `adjust!`?

File: docs/src/index.md [hunk 1]
```diff
@@ -165,6 +165,26 @@ Optimisers.thaw!(opt)
 opt.layers[3].bias  # Leaf(Momentum(...), [0.0, 0.0])
 ```
 
+## Adjusting Learning Rate
```
Comment (ToucheSir) [inline docs/src/index.md:168]:
```suggestion
## Adjusting Hyperparameters
```
Or something similar. The thought being that this is more searchable.

File: test/runtests.jl [hunk 2]
```diff
@@ -177,7 +177,7 @@ end
       @test eltype(s6[2].state[2]) == Float32
     end
 
-    @testset "adjusyting parameters" begin
+    @testset "adjusyting parameters, 1" begin
```
Comment (ToucheSir) [inline test/runtests.jl:180]:
```suggestion
    @testset "adjusting parameters, out-of-place" begin
```

File: test/runtests.jl [hunk 3]
```diff
@@ -221,6 +221,50 @@ end
       @test sc2.γ.state[2][1] ≈ [0.1, 0.2, 0.2]
     end
 
+    @testset "adjusyting parameters, 2!" begin  # Same tests, just in-place
```
Comment (ToucheSir) [inline test/runtests.jl:224]:
```suggestion
    @testset "adjusting parameters, in-place" begin
```

**Review by ToucheSir — APPROVED** [review]
